### ISSUE TITLE
implement on my dataset？

### ISSUE THREAD

**Issue by wamm0909** [AlexZaikin94/MoCo-v2#1]
can‘i use this code run my dataset？where should i change，and please give me some suggestions，ths

**Comment by AlexZaikin94** [comment]
this repo is designed to run with the Imagenette dataset, so any data that is structured in the same way should run with no adjustments (except perhaps minor adjustments in config.py).
you can also change the Dataset class in src/utils.py for your needs.

**Comment by AlexZaikin94** [comment]
> can‘i use this code run my dataset？where should i change，and please give me some suggestions，ths

this repo is designed to run with the Imagenette dataset, so any data that is structured in the same way should run with no issues (perhaps minor adjustments may be needed in config.py).
you can also change the Dataset class in src/utils.py for your needs.

**Comment by wamm0909** [comment]
> > can‘i use this code run my dataset？where should i change，and please give me some suggestions，ths
> 
> this repo is designed to run with the Imagenette dataset, so any data that is structured in the same way should run with no issues (perhaps minor adjustments may be needed in config.py).
> you can also change the Dataset class in src/utils.py for your needs.

ths，respect

**Comment by wamm0909** [comment]
> > can‘i use this code run my dataset？where should i change，and please give me some suggestions，ths
> 
> this repo is designed to run with the Imagenette dataset, so any data that is structured in the same way should run with no issues (perhaps minor adjustments may be needed in config.py).
> you can also change the Dataset class in src/utils.py for your needs.

buddy, can this demo achieve the result shown in the figure below?
![1231630915811_ pic](https://user-images.githubusercontent.com/38876426/132655692-3923f2e1-921e-46a9-b4c4-45f6d9f53bbd.jpg)

**Comment by AlexZaikin94** [comment]
> > > can‘i use this code run my dataset？where should i change，and please give me some suggestions，ths
> > 
> > 
> > this repo is designed to run with the Imagenette dataset, so any data that is structured in the same way should run with no issues (perhaps minor adjustments may be needed in config.py).
> > you can also change the Dataset class in src/utils.py for your needs.
> 
> buddy, can this demo achieve the result shown in the figure below?
> ![1231630915811_ pic](https://user-images.githubusercontent.com/38876426/132655692-3923f2e1-921e-46a9-b4c4-45f6d9f53bbd.jpg)

i'm not sure what you mean by achieving the results in the picture.
the picture looks like an ilustration of contrastive learning, this repo is an implementation of the [MoCo v1: Momentum Contrastive learning](https://arxiv.org/abs/1911.05722) paper (and v2) which also does pretraining via contrastive learning but with momentum, meaning there is a 2nd model (called the "key model"), which is gradually updated using the main model (called "query model"), such that in each pair of images (as in your ilustration) the 1st is embeded using the "query model " and the 2nd is using embeded the "key model".
as a result of the unsupervised training process, representations of different augmentations of the same image will hopefully be closer, and representations of different images will be further apart.

**Comment by wamm0909** [comment]
> > > > can‘i use this code run my dataset？where should i change，and please give me some suggestions，ths
> > > 
> > > 
> > > this repo is designed to run with the Imagenette dataset, so any data that is structured in the same way should run with no issues (perhaps minor adjustments may be needed in config.py).
> > > you can also change the Dataset class in src/utils.py for your needs.
> > 
> > 
> > buddy, can this demo achieve the result shown in the figure below?
> > ![1231630915811_ pic](https://user-images.githubusercontent.com/38876426/132655692-3923f2e1-921e-46a9-b4c4-45f6d9f53bbd.jpg)
> 
> i'm not sure what you mean by achieving the results in the picture.
> the picture looks like an ilustration of contrastive learning, this repo is an implementation of the [MoCo v1: Momentum Contrastive learning](https://arxiv.org/abs/1911.05722) paper (and v2) which also does pretraining via contrastive learning but with momentum, meaning there is a 2nd model (called the "key model"), which is gradually updated using the main model (called "query model"), such that in each pair of images (as in your ilustration) the 1st is embeded using the "query model " and the 2nd is using embeded the "key model".
> as a result of the unsupervised training process, representations of different augmentations of the same image will hopefully be closer, and representations of different images will be further apart.

ths buddy，I understand your reply. What I mean is that moco demo can gather similar images in a folder（may be i add  a clustering method to moco to achieve）? like as ilustration picture similar ↑

**Comment by AlexZaikin94** [comment]
> > > > > can‘i use this code run my dataset？where should i change，and please give me some suggestions，ths
> > > > 
> > > > 
> > > > this repo is designed to run with the Imagenette dataset, so any data that is structured in the same way should run with no issues (perhaps minor adjustments may be needed in config.py).
> > > > you can also change the Dataset class in src/utils.py for your needs.
> > > 
> > > 
> > > buddy, can this demo achieve the result shown in the figure below?
> > > ![1231630915811_ pic](https://user-images.githubusercontent.com/38876426/132655692-3923f2e1-921e-46a9-b4c4-45f6d9f53bbd.jpg)
> > 
> > 
> > i'm not sure what you mean by achieving the results in the picture.
> > the picture looks like an ilustration of contrastive learning, this repo is an implementation of the [MoCo v1: Momentum Contrastive learning](https://arxiv.org/abs/1911.05722) paper (and v2) which also does pretraining via contrastive learning but with momentum, meaning there is a 2nd model (called the "key model"), which is gradually updated using the main model (called "query model"), such that in each pair of images (as in your ilustration) the 1st is embeded using the "query model " and the 2nd is using embeded the "key model".
> > as a result of the unsupervised training process, representations of different augmentations of the same image will hopefully be closer, and representations of different images will be further apart.
> 
> ths buddy，I understand your reply. What I mean is that moco demo can gather similar images in a folder（may be i add a clustering method to moco to achieve）? like as ilustration picture similar ↑

i did not implement/plan to implement that feature, and in fact this repo does not actually help you with that...
all you need is to extract the image representations from your data (using any pre-trained model you like, pytorch has a rich variety of pre-trained models) and use a clustering algorithm on the representations, you can then compute the similarity with a simple dot product (the same basic similarity metric used in MoCo) between the representations of each pair of images.

**Comment by wamm0909** [comment]
> > > > > > can‘i use this code run my dataset？where should i change，and please give me some suggestions，ths
> > > > > 
> > > > > 
> > > > > this repo is designed to run with the Imagenette dataset, so any data that is structured in the same way should run with no issues (perhaps minor adjustments may be needed in config.py).
> > > > > you can also change the Dataset class in src/utils.py for your needs.
> > > > 
> > > > 
> > > > buddy, can this demo achieve the result shown in the figure below?
> > > > ![1231630915811_ pic](https://user-images.githubusercontent.com/38876426/132655692-3923f2e1-921e-46a9-b4c4-45f6d9f53bbd.jpg)
> > > 
> > > 
> > > i'm not sure what you mean by achieving the results in the picture.
> > > the picture looks like an ilustration of contrastive learning, this repo is an implementation of the [MoCo v1: Momentum Contrastive learning](https://arxiv.org/abs/1911.05722) paper (and v2) which also does pretraining via contrastive learning but with momentum, meaning there is a 2nd model (called the "key model"), which is gradually updated using the main model (called "query model"), such that in each pair of images (as in your ilustration) the 1st is embeded using the "query model " and the 2nd is using embeded the "key model".
> > > as a result of the unsupervised training process, representations of different augmentations of the same image will hopefully be closer, and representations of different images will be further apart.
> > 
> > 
> > ths buddy，I understand your reply. What I mean is that moco demo can gather similar images in a folder（may be i add a clustering method to moco to achieve）? like as ilustration picture similar ↑
> 
> i did not implement/plan to implement that feature, and in fact this repo does not actually help you with that...
> all you need is to extract the image representations from your data (using any pre-trained model you like, pytorch has a rich variety of pre-trained models) and use a clustering algorithm on the representations, you can then compute the similarity with a simple dot product (the same basic similarity metric used in MoCo) between the representations of each pair of images.

ths buddy, your reply help me a lot. respect
